### PR TITLE
Invites: Improves translated string for invite header

### PR DESCRIPTION
**Pull request by ebinnion**
Addresses part of #138 and related to #369

To test:
- Checkout `update/invite-header-role-strings` branch
- Invite a test user to join a WordPress.com site by going to `Users > Invite New` in `wp-admin`
- In the invitation email, make note of the invitation key
- Go to `/accept-invite/$site/$invite_key`
- Does the "invited you..." text seem appropriate for the role?
- Now, go back to the invite users UI in `wp-admin`
- Invite the same user, but with a different role this time
- Refresh the tab that has Calypso loaded
- Do the strings represent the new role?

cc @roccotripaldi for review and @rickybanister for wording review

### CODE REVIEW

File: client/accept-invite/invite-header/index.jsx [hunk 1]
```diff
@@ -16,15 +16,90 @@ import Gravatar from 'components/gravatar';
...
+					'{{inviterName/}} invited you to edit:', {
+						components: {
+							inviterName: inviterName
+						}
+					}
+				);
 				break;
+			case 'author':
+				text = this.translate(
+					'{{inviterName/}} invited you to author on:', {
```
Comment (beaulebens) [inline client/accept-invite/invite-header/index.jsx]: `invited you to be an author on:`

**Comment by rickybanister** [comment]
They all seem great except this is weird:

`invited you to author on`

Let's go with 

`invited you to be an author on:`

It doesn't match the others if we don't use 'author' as a verb, but I don't feel it's commonly used as a verb these days.

File: client/accept-invite/invite-header/index.jsx [hunk 2]
```diff
@@ -16,15 +16,90 @@ import Gravatar from 'components/gravatar';
...
+		const inviterName = (
+			<strong className="invite-header__inviter-name">
+				{ this.getInviterName() }
+			</strong>
+		);
+
+		switch ( get( this.props, 'invite.meta.role' ) ) {
+			case 'administrator':
+				text = this.translate(
+					'{{inviterName/}} invited you to administer:', {
```
Comment (roccotripaldi) [inline client/accept-invite/invite-header/index.jsx]: `User invited you to administer:` feels a bit strange. I'd prefer `User invited you to manage:`

**Comment by roccotripaldi** [comment]
Code looks great, and works as expected. I left one minor wording fix.

**Comment by rickybanister** [comment]
It appears this is already merged, but I don't think we should have switched to use the word 'manage'.

The role is 'admin' so I think using some derivation of that term is important. 

`has invited you to be an admin on:` would be better than `invited you to manage` I think.

**Pull request merged**